Tasks/BeanstalkDeployApplication/helpers/DeployApplicationTaskOperations.ts

```diff
@@ -163,36 +168,54 @@ export class TaskOperations {
 
         let success = true;
         let environment: Beanstalk.EnvironmentDescription;
+
+        // delay the event poll by a random amount, up to 5 seconds, so that if multiple
+        // deployments run in parallel they don't all start querying at the same time and
+        // potentially trigger throttling from the outset
+        const initialStartDelay = Math.floor(Math.random() * (randomJitterUpperLimit)) + 1;
+        await this.sleep(initialStartDelay * 1000);
+
         do {
             tl.debug(`...event poll sleep for ${eventPollDelay}s`);
             await this.sleep(eventPollDelay * 1000);
 
-            const responseEnvironments = await this.beanstalkClient.describeEnvironments(requestEnvironment).promise();
-            if (responseEnvironments.Environments.length === 0) {
-                throw new Error(tl.loc('FailedToFindEnvironment'));
-            }
-            environment = responseEnvironments.Environments[0];
+            // if any throttling exception escapes the sdk's default retry logic,
+            // extend the user's selected poll delay by a random, sub-5 second, amount
+            try {
+                const responseEnvironments = await this.beanstalkClient.describeEnvironments(requestEnvironment).promise();
+                if (responseEnvironments.Environments.length === 0) {
+                    throw new Error(tl.loc('FailedToFindEnvironment'));
+                }
+                environment = responseEnvironments.Environments[0];
 
-            requestEvents.StartTime = lastPrintedEventDate;
-            const responseEvent = await this.beanstalkClient.describeEvents(requestEvents).promise();
+                requestEvents.StartTime = lastPrintedEventDate;
+                const responseEvent = await this.beanstalkClient.describeEvents(requestEvents).promise();
 
-            if (responseEvent.Events.length > 0) {
-                for (let i = responseEvent.Events.length - 1; i >= 0; i--) {
-                    const event = responseEvent.Events[i];
-                    if (event.EventDate <= lastPrintedEventDate) {
-                        continue;
-                    }
+                if (responseEvent.Events.length > 0) {
```

Comment:
1. `if (responseEvent.Events.length > 0)` is redundant, remove this check.
2. This whole loop can be simplified:

```typescript
for (const event of responseEvent.Events.filter(e => e.EventDate <= lastPrintedEventDate)) {
    console.log(event.EventDate + '   ' + event.Severity + '   ' + event.Message)

    if (event.Message === 'Failed to deploy application.') {
        success = false
    }
}
```

Reply:
sure we can do that refactor, these kinds of things need to be fixed eventually